### PR TITLE
add LAION 2B only training

### PR DESCRIPTION
**Pull request by Soonhwan-Kwon**
add LAION 2B only training related to the issue below
https://github.com/mlfoundations/open_flamingo/issues/137

### CODE REVIEW

**Comment by Soonhwan-Kwon** [comment]
it is strange when I run psf/black on my machine, it outputs
```
python -m black open_flamingo/
reformatted /data4/open_flamingo/open_flamingo/train/train_utils.py

All done! ✨ 🍰 ✨
1 file reformatted, 19 files left unchanged.
```

**Comment by anas-awadalla** [comment]
Can you upgrade black? I remember I ran into this problem earlier and that was the fix. If not no worries! Thanks for putting this together. I will take a look and merge :)

File: open_flamingo/train/train_utils.py [hunk 1]
```diff
@@ -64,193 +67,314 @@ def train_one_epoch(
...
+                    )
+                    step_time_m.reset()
+                    data_time_m.reset()
+
+                    wandb.log(
+                        {
+                            "loss_laion": loss.item(),
+                            "global_step": global_step,
+                        },
+                        commit=False,
```
Comment (anas-awadalla) [inline open_flamingo/train/train_utils.py:370]: commit should be true for the last wandb log in an iteration

File: open_flamingo/train/train_utils.py [hunk 2]
```diff
@@ -64,193 +67,314 @@ def train_one_epoch(
...
-
-        for i in range(labels.shape[0]):
-            # remove loss for any token before the first <image> token
-            label_idx = 0
-            while (
-                label_idx < labels.shape[1] and labels[i][label_idx] != media_token_id
-            ):
-                labels[i][label_idx] = -100
-                label_idx += 1
+    if mmc4_loader is not None:
```
Comment (anas-awadalla) [inline open_flamingo/train/train_utils.py:70]: I do not like the if/else approach. There is a lot of duplicate code that will make it very difficult to maintain this.

Comment (anas-awadalla) [reply open_flamingo/train/train_utils.py:70]: Can you please refactor this to remove any duplicate code? For example the forward pass from laion, the code for masking out gradients in the embedding, iteration through dataloaders, wandb logging all have a bit of repetition.

**Review by anas-awadalla — CHANGES_REQUESTED** [review]
Thanks for putting this together. I think we should first clean up the duplicate code in train_utils.py before merging.

File: open_flamingo/train/train_utils.py (hunk 2 above)
Comment (Soonhwan-Kwon) [reply open_flamingo/train/train_utils.py:70]: sure, I wrote some code already, I'll make PR in short.

File: open_flamingo/train/train_utils.py (hunk 1 above)
Comment (Soonhwan-Kwon) [reply open_flamingo/train/train_utils.py]: Thany you for your feedback, I fixed it in next commit.

**Comment by Soonhwan-Kwon** [comment]
I committed updates last week, and I tested run for laion dataset. Please let me know if there are more feedbacks.

**Comment by anas-awadalla** [comment]
Thanks @Soonhwan-Kwon! We will merge this but may take a few days for me to do so.
